opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -110,9 +110,11 @@ private String processSprayEvent(Event event, ClientClassification clientClassif
             if (task != null) {
                 task.setBusinessStatus(calculateBusinessStatus(event));
                 task.setStatus(Task.TaskStatus.COMPLETED);
-                if (localEvents) {
+                //update task status to unsynced if it was synced, ignore if task status is created so that it will be created on server
+                if (localEvents && BaseRepository.TYPE_Synced.equals(task.getSyncStatus())) {
```

Comment:
What happens when events with the status created hit the server?

Why is it necessary to again sync these tasks?

Reply:
Events doesn't have Created status but Tasks do. So am not sure of the question.

The flow is a spray event is generated after saving a spray form. During client processing on the device it updates the task status locally. This task needs to be synced to server so that it spray can be reflected on mspray dashboards.

Updating the task status only updates the task status on the server, it does not override task data. 

However if the task was never synced to server e.g a structures added and sprayed immediately the task should the created on the server and not updated.